MiniProject 1/VideoViewController.swift

```diff
@@ -0,0 +1,115 @@
+    var urls = ["http://fun.siz.io/stories/142893791787803c7fb48f4d/0.mp4",
+                "http://fun.siz.io/stories/142893791787803c7fb48f4d/1.mp4",
+                "http://fun.siz.io/stories/142893791787803c7fb48f4d/2.mp4",
+                "http://fun.siz.io/stories/142893791787803c7fb48f4d/3.mp4"]
+    
+    var videoNumber = 0
+    
+    func displayButton()
+    {
+        let leftButton = UIButton.buttonWithType(UIButtonType.System) as! UIButton
+        leftButton.frame = CGRectMake(0, (screenSize.width/2)-20, 80, 40)
+        leftButton.backgroundColor = .blackColor()
+        leftButton.setTitle("Prev", forState: UIControlState.Normal)
+        leftButton.setTitleColor(UIColor.whiteColor(), forState: UIControlState.Normal)
+        leftButton.addTarget(self, action: "leftButtonAction:", forControlEvents: UIControlEvents.TouchUpInside)
+        
+        let rightButton = UIButton.buttonWithType(UIButtonType.System) as! UIButton
+        rightButton.frame = CGRectMake((screenSize.height)-80, (screenSize.width/2)-20, 80, 40)
+        rightButton.backgroundColor = .blackColor()
+        rightButton.setTitle("Next", forState: UIControlState.Normal)
+        rightButton.setTitleColor(UIColor.whiteColor(), forState: UIControlState.Normal)
+        rightButton.addTarget(self, action: "rightButtonAction:", forControlEvents: UIControlEvents.TouchUpInside)
+        
+        self.view.addSubview(leftButton)
+        self.view.addSubview(rightButton)
+    }
+    
+    func leftButtonAction(sender:UIButton!)
+    {
+        moviePlayer!.pause()
+        
+        if(videoNumber > 0)
+        {
+            videoNumber--
+        }
+        else if(videoNumber == 0)
+        {
+            videoNumber = urls.count-1
+        }
+        
+        playVideo()
+    }
+    
+    func rightButtonAction(sender:UIButton!)
+    {
+        moviePlayer!.pause()
+        
+        if(videoNumber < urls.count-1)
+        {
+            videoNumber++
+        }
+        else if(videoNumber == urls.count-1)
+        {
+            videoNumber = 0
+        }
+        
+        playVideo()
+    }
+    
+    func playVideo()
+    {
+        var url = NSURL(string: urls[videoNumber])
+        
+        playerItem = AVPlayerItem(URL: url)
+        moviePlayer = AVPlayer(playerItem: playerItem)
+        playerLayer = AVPlayerLayer(player: moviePlayer)
+        moviePlayer!.actionAtItemEnd = .None
+        
+        playerLayer!.frame = CGRect(x: 0, y: 0, width: screenSize.height, height: screenSize.width)
+        //playerLayer!.backgroundColor = UIColor.blackColor().CGColor
+        self.view.backgroundColor = .blackColor()
+        self.view.layer.addSublayer(playerLayer)
+        
+        moviePlayer!.play()
+        displayButton()
+        
+        NSNotificationCenter.defaultCenter().addObserver(self,
+            selector: "restartVideo",
+            name: AVPlayerItemDidPlayToEndTimeNotification,
+            object: moviePlayer!.currentItem)
```

Comment:
You have to cleanup the notification when the object "moviePlayer!.currentItem" or self is destroyed.